### PR TITLE
[MIG] website_mail_channel: migration to new api.

### PR DESCRIPTION
**Pull request by kga-odoo**
<strong>Task:</strong>https://www.odoo.com/web#id=12029&view_type=form&model=project.task&action=327
<strong>Pad:</strong>https://pad.odoo.com/p/r.Yk2pVlLvxdw8HYUr

### CODE REVIEW

File: addons/website_mail_channel/__openerp__.py [hunk 1]
```diff
@@ -10,7 +10,7 @@
         """,
     'depends': ['website_mail'],
     'data': [
-        'views/website_mail_channel.xml',
+        'views/mail_channel_views.xml',
```
Comment (rgo-odoo) [inline addons/website_mail_channel/__openerp__.py]: It should be renamed to mail_channel_templates.xml

File: addons/website_mail_channel/controllers/main.py [hunk 2]
```diff
@@ -3,10 +3,9 @@
 import datetime
```
Comment (rgo-odoo) [inline addons/website_mail_channel/controllers/main.py]: you may import openerp.fields' Date and Datetime classes and use from_string/to_string functions of theirs to perform date operations.

File: addons/website_mail_channel/controllers/main.py [hunk 3]
```diff
@@ -52,12 +42,11 @@ def view(self, **post):
     @http.route(["/groups/subscription/"], type='json', auth="user")
     def subscription(self, group_id=0, action=False, **post):
         """ TDE FIXME: seems dead code """
-        cr, uid, context = request.cr, request.uid, request.context
-        group_obj = request.registry.get('mail.channel')
+        Group = request.env['mail.channel']
         if action:
-            group_obj.message_subscribe_users(cr, uid, [group_id], context=context)
+            Group.message_subscribe_users(user_ids=[group_id])
```
Comment (rgo-odoo) [inline addons/website_mail_channel/controllers/main.py]: Call functions message_subscribe_users/message_unsubscribe_users  with "mail.channel" recordset of "group_id".

File: addons/website_mail_channel/controllers/main.py [hunk 4]
```diff
@@ -100,20 +86,13 @@ def thread_headers(self, group, page=1, mode='thread', date_begin=None, date_end
...
             base_domain = [('model', '=', 'mail.channel'), ('res_id', '=', group.id)]
-        next_message = None
-        next_message_ids = Message.search(cr, uid, base_domain + [('date', '<', message.date)], order="date DESC", limit=1, context=context)
-        if next_message_ids:
-            next_message = Message.browse(cr, uid, next_message_ids[0], context=context)
-        prev_message = None
-        prev_message_ids = Message.search(cr, uid, base_domain + [('date', '>', message.date)], order="date ASC", limit=1, context=context)
-        if prev_message_ids:
-            prev_message = Message.browse(cr, uid, prev_message_ids[0], context=context)
+        next_message = Message.search(base_domain + [('date', '<', message.date)], order="date DESC", limit=1) or None
```
Comment (rgo-odoo) [inline addons/website_mail_channel/controllers/main.py]: "or" condition not needed.

File: addons/website_mail_channel/models/mail_mail.py [hunk 5]
```diff
@@ -1,24 +1,24 @@
 # -*- coding: utf-8 -*-
 
-from openerp.osv import osv
-from openerp import tools
+from openerp import api, models, tools
 from openerp.tools.translate import _
```
Comment (rgo-odoo) [inline addons/website_mail_channel/models/mail_mail.py]: import _ with above statement

File: addons/website_mail_channel/controllers/main.py [hunk 6]
```diff
@@ -14,47 +14,35 @@ class MailGroup(http.Controller):
...
-        return request.website.render('website_mail_channel.mail_channels', values)
+        return request.website.render('website_mail_channel.mail_channels', {'groups': groups, 'group_data': group_data})
 
     @http.route(["/groups/is_member"], type='json', auth="public", website=True)
     def is_member(self, channel_id=0, **kw):
         """ Determine if the current user is member of the given channel_id
             :param channel_id : the channel_id to check
         """
-        current_user_id = request.uid
+        current_user_id = request.env.uid
```
Comment (hmo-odoo) [inline addons/website_mail_channel/controllers/main.py]: can you refactor line:51 ? I think, no need to browse current user record. use request.env.user.partner_id

File: addons/website_mail_channel/controllers/main.py [hunk 7]
```diff
@@ -88,15 +76,15 @@ def subscription(self, channel_id=0, subscription="on", email='', **kw):
         partner_ids = []
 
         # search partner_id
-        current_user_id = request.uid
+        current_user_id = request.env.uid
```
Comment (hmo-odoo) [inline addons/website_mail_channel/controllers/main.py]: line:82 also. use request.env.user.partner_id

File: addons/website_mail_channel/controllers/main.py [hunk 8]
```diff
@@ -88,15 +76,15 @@ def subscription(self, channel_id=0, subscription="on", email='', **kw):
...
         public_id = request.website.user_id.id
         if current_user_id != public_id:
-            partner_ids = [request.env['res.users'].browse(current_user_id).partner_id.id]
+            partner_ids = request.env['res.users'].browse(current_user_id).partner_id.ids
```
Comment (hmo-odoo) [inline addons/website_mail_channel/controllers/main.py]: request.env.user.partner_id

**Comment by tde-banana-odoo** [comment]
Work continues on https://github.com/odoo/odoo/pull/8163 . Thanks !

**Pull request closed without merging**